src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.ts

```diff
@@ -137,22 +187,95 @@ export class FlowsTableComponent implements OnInit, OnChanges {
                 noButtonText: "Cancel",
                 handler: (ok) => {
                     if (ok) {
-                        this.cancelFlowChange.emit(flowId);
+                        this.cancelFlowChange.emit({ flowId, datasetId });
                     }
                 },
             }),
         );
     }
 
-    public navigateToFlowDetaisView(flowId: string): void {
+    public navigateToFlowDetaisView(flow: FlowSummaryDataFragment, datasetId: string): void {
         this.navigationService.navigateToFlowDetails({
-            accountName: this.datasetBasics.owner.accountName,
-            datasetName: this.datasetBasics.name,
-            flowId,
+            accountName: this.accountView && this.accountName ? this.accountName : this.datasetBasics.owner.accountName,
+            datasetName: this.accountView ? this.datasetByIdForAccount(datasetId).name : this.datasetBasics.name,
+            flowId: flow.flowId,
         });
     }
```

Comment:
This is a proof having separate inputs per mode is unacceptable. it should be a simple handler that works universally.

Reply:
Removed, created universal port object